packages/core/src/components/way-autosuggest/way-autosuggest.tsx

```diff
@@ -0,0 +1,253 @@
+import { Component, Method, Element, Event, EventEmitter, Listen, State, Prop, Watch, h } from '@stencil/core';
+import { IFormElementData } from '../../types/form';
+import { ISelectOption, ISelectConfig } from '../../types/select';
+
+@Component({
+  tag: 'way-autosuggest',
+  styleUrl: 'way-autosuggest.scss',
+  shadow: true,
+})
+export class W2wSelect {
+  private optionListEl?: HTMLInputElement;
+  private inputEl?: HTMLInputElement;
+  @Element() el!: HTMLWayAutosuggestElement;
+
+  @Prop() options?: Array<ISelectOption> = [];
+  @Prop() config?: ISelectConfig = {};
+  @Prop() validation?: (value: any) => string[];
+  @Prop() value?: string | Array<{ label: string; value: any }>;
+  @Prop() name: string;
+
+  @State() inputValue: string = '';
+  @State() hasFocus = false;
+  @State() filteredOptions: Array<ISelectOption> = this.options;
+  @State() localSelected = [];
+  @State() localConfig: ISelectConfig = {
+    selectedText: 'selected',
+    maxTagWidth: '10rem',
+    tagColor: 'green',
+    ...this.config,
+  };
+  @State() highlightIndex: number = 0;
+
+  // use until clicking outside works
+  @State() scrollPos: number = 0;
+
+  @Method()
+  async getValue() {
+    return this.localSelected;
+  }
+
+  @Listen('click', { target: 'window' })
```

Comment:
@elwinvaneede suggested to replace this with a listener on a a (hidden) overlay.

I agree with that suggestion but would like to postpone this to a later stage due to the limited time I can work on this.

Like we all discussed in one of the first FE meetings: It is maybe a good idea to at least have the working basics before making everything perfect. I would like that to apply to this specific piece of code

Reply:
Yes, agreed